plugins/carbon_black_cloud/icon_carbon_black_cloud/tasks/monitor_alerts/task.py

```diff
@@ -209,9 +232,21 @@ def get_observations(
                         f"More data is available on the API (num_found={num_found}) - setting has_more_pages=True..."
                     )
                     has_more_pages = True
+                else:
+                    end_of_window = state.get(OBSERVATION_END_TIME)
+                    has_more_pages = datetime.strptime(end_of_window, TIME_FORMAT).replace(tzinfo=timezone.utc) != now
+            else:
+                state[LAST_OBSERVATION_TIME] = state.get(OBSERVATION_END_TIME)
+                has_more_pages = True
+
             # remove the job ID as this is completed and next run we want to trigger a new one
             del state[LAST_OBSERVATION_JOB]
             del state[LAST_OBSERVATION_JOB_TIME]
+            if state.get(LAST_OBSERVATION_TIME) is start_observation_time:
+                state[OBSERVATION_JOB_OFFSET] = state.get(OBSERVATION_JOB_OFFSET, 0) + page_size
```

Comment:
would it not make sense to include the offset bump in the `num_found > page_size` condition, followed by a subsequent condition that checks the last observation time hasn't moved? `observations[-1][<time_field>] != observations[0][<time_field>]` or compare against `state[OBSERVATION_END_TIME]`?

Reply:
We chatted on a call through some of these things.